cmd/net-crd-installer/utils/net_crd_installer_integration_test.go

```diff
@@ -0,0 +1,194 @@
+		Expect(len(spec.Properties)).Should(Equal(4), "CRD %s should have 4 properties defined in spec", crdName)
+		Expect(spec.Properties["bar"].Type).Should(Equal("string"), "CRD %s should have 'bar' property of type string defined in properties", crdName)
+		_, ok := spec.Properties["newField"]
+		Expect(ok).To(BeFalse(), "CRD %s should not have 'newField' property defined in properties", crdName)
+	})
+
+	It("update the CRD to add a random label", func() {
+		updateCRDLabels(crdName, map[string]string{randomLabelKey: "true"})
+	})
+
+	It("should update the CRD with new field in spec with crdinstaller label", func() {
+		crd, err := getCRDFromPath(updatedCRDPath, scheme)
+		Expect(err).NotTo(HaveOccurred(), "should get CRD from path %s", updatedCRDPath)
+		Expect(InstallCRD(ctx, k8sClient, crd, true)).To(Succeed())
+	})
+
+	It("should verify updated CRD", func() {
+		// ensure we don't overwrite the random label.
+		ensureCRDExistsWithLabels(map[string]string{
+			randomLabelKey:       "true",
+			crdInstallerLabelKey: "true",
+			// Azure managed label should not be set in E2E tests.
+		})
+		crd := &apiextensionsv1.CustomResourceDefinition{}
+		Expect(k8sClient.Get(ctx, types.NamespacedName{Name: crdName}, crd)).NotTo(HaveOccurred(), "CRD %s should be installed", crdName)
+		spec := fetchSpecJSONSchemaProperties(crd)
+		// Updated CRD should have 5 properties defined in spec.
+		Expect(len(spec.Properties)).Should(Equal(5), "CRD %s should have 5 properties defined in spec", crdName)
+		Expect(spec.Properties["bar"].Type).Should(Equal("string"), "CRD %s should have 'bar' property of type string defined in properties", crdName)
+		_, ok := spec.Properties["newField"]
+		Expect(ok).To(BeTrue(), "CRD %s should have 'newField' property defined in properties", crdName)
+		Expect(spec.Properties["newField"].Type).Should(Equal("string"), "CRD %s should have 'newField' property of type string defined in properties", crdName)
+	})
+
+	It("should delete the CRD", func() {
+		crd := &apiextensionsv1.CustomResourceDefinition{}
+		Expect(k8sClient.Get(ctx, types.NamespacedName{Name: crdName}, crd)).NotTo(HaveOccurred(), "CRD %s should exist before deletion", crdName)
+		Expect(k8sClient.Delete(ctx, crd)).To(Succeed(), "CRD %s should be deleted successfully", crdName)
+
+		Eventually(func() bool {
+			return apierrors.IsNotFound(k8sClient.Get(ctx, types.NamespacedName{Name: crdName}, crd))
+		}, eventuallyDuration, eventuallyInterval).Should(BeTrue(), "CRD %s should be deleted", crdName)
+	})
+})
+
+// This test verifies the behavior of the CRD installer when E2E mode is set to false.
+// It ensures that the installer can create a CRD, update it with new fields, and handle ownership label correctly.
+// The original CRD has 4 properties, and the updated CRD has a new property to simulate CRD upgrade.
+var _ = Describe("Test CRD Installer, Create and Update CRD, E2E mode set to false", Ordered, func() {
```

Comment:
A lot of duplications here. The tests are probably going to be removed/reduced after we move to use fleet lib. Next PR.